docs/global/modules/core/pages/reference/component-model-api-gen-gradle.adoc

```diff
@@ -100,7 +100,12 @@ Inside of the `metamodel` block the following settings can be configured.
 
 |`registrationHelperName`
 |String
-|Name of the registration helper
+|Name of the generated language registration helper
+
+|`conceptPropertiesInterfaceName`
+|String
+|Name of the generated interface, that contains the concept meta-properties of this language set.
+If null (default), neither the concept meta-properties nor the corresponding interface will be generated.
```

Comment:
Is there any overhead in generating one more interface? Or why do we want to include more conditional logic in the whole generation process? Isn't it more stuff to maintain if we make this conditional?

Reply:
This is mainly for backwards compatibility, so that generation doesn't break for existing users if no name is specified.

Reply:
There could simply be a reasonable default?

Reply:
Ah, what about putting that interface into the model-api library itself instead of generating it?

Reply:
Similar to that enum base class/interface ;-)

Reply:
Because the meta properties are not known in advance. From MPS we only export the alias by default. But the user can add arbitrary meta properties in the ConceptData json, which will be generated into the interface.

Reply:
I don't quite understand the reasoning. For enums we generically know that there's an "presentation", but for concepts we cannot assume that there's an optional "alias". Both things sound pretty similar to me.

Reply:
Enums (including their presentation) were added to the json schema. Since the concept alias has no semantic meaning and is too MPS specific, we did not want to add it to the schema directly but still support it. That's why this PR adds the extremely generic metaInfo map for meta properties to concepts. Our export from MPS adds the alias meta property, but the generator does not know about this.
We do not know, which properties the user puts in there. That's why the interface must be generated.

Reply:
https://issues.modelix.org/issue/MODELIX-555/Support-meta-properties-on-all-meta-model-elements